### PR TITLE
updates subscribe method

### PR DESCRIPTION
**Pull request by Nathan-Notifi**
https://user-images.githubusercontent.com/105258726/196400654-e848c436-eaf8-445c-9572-8230815c85bb.mov

### CODE REVIEW

File: packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts [hunk 1]
```diff
@@ -279,37 +299,38 @@ export const useNotifiSubscribe: () => Readonly<{
...
+
+      for (let i = 0; i < alerts.length; i++) {
+        const existingAlert = alerts[i];
+        console.log('existing alert', existingAlert);
         if (
-          source === undefined ||
-          source.id === null ||
-          filter === undefined ||
-          filter.id === null
+          existingAlert !== undefined &&
```
Comment (Nathan-Notifi) [inline packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts:323]: it gets funky here. i create the alerts for configurations but if not, i attempt to update all existing alerts

File: packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts [hunk 2]
```diff
@@ -279,37 +299,38 @@ export const useNotifiSubscribe: () => Readonly<{
           );
         }
 
+        const alert = await client.createAlert({
+          emailAddress: finalEmail,
+          filterId: filter?.id ?? '',
```
Comment (Nathan-Notifi) [inline packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts:304]: how should i handle these?

File: packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts [hunk 3]
```diff
@@ -279,37 +299,38 @@ export const useNotifiSubscribe: () => Readonly<{
...
+          targetGroupName: 'Default',
+          telegramId: finalTelegramId,
+        });
+
+        newResults[currentConfigurationData.alertName] = alert;
+      }
+    } else {
+      const alerts = data.alerts;
+
+      for (let i = 0; i < alerts.length; i++) {
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts:319]: this part is not actually sufficient

when we have alerts, we need to:
- identify which alertConfigurations have a corresponding alert
  - if the corresponding alert is the same (i.e. same source, same target) do not do anything
  - if the corresponding alert is not the same, delete it and recreate it using configuration
- identify which alerts have a `null` alertConfiguration and delete them
- if there exists an alert whose name doesn't appear in alertConfiguration, leave them alone

happy to talk through what i mean

File: packages/notifi-react-card/lib/context/NotifiSubscriptionContext.tsx [hunk 4]
```diff
@@ -58,6 +64,7 @@ export const NotifiSubscriptionContextProvider: React.FC<
   PropsWithChildren<NotifiParams>
 > = ({ children, ...params }) => {
   const [email, setEmail] = useState<string>('');
+  const [eventTypes, setEventTypes] = useState<EventTypeConfig>([]);
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/context/NotifiSubscriptionContext.tsx:67]: oh hm let's not leak this all the way down

possible to call setAlertConfiguration instead from the FetchedCard? using eventTypes mapped to alert configurations

**Review by juni-notifi — COMMENTED** [review]
some quick thoughts, lmk

File: packages/notifi-react-card/lib/utils/AlertConfiguration.ts [hunk 5]
```diff
@@ -1,7 +1,10 @@
+// import { AlertConfiguration } from './AlertConfiguration';
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/utils/AlertConfiguration.ts:1]: ?

**Review by juni-notifi — APPROVED** [review]

**Pull request merged**